packages/api-tests/test/window.spec.js

```diff
@@ -106,6 +137,254 @@ describe('Window API', function(done) {
+        (result) => assert.equal(result.value.length, 1)
+      ];
+
+      return chainPromises(steps);
+    });
+
+    it('Should not get child windows if there isn\'t any', function() {
+      const windowTitle = 'windownameclose';
+      const windowOptions = getWindowOptions({
+        name: windowTitle
+      });
+
+      const steps = [
+        ...setupWindowSteps(windowOptions),
+        () => selectWindow(app.client, 1),
+        () => callWindowMethod('getChildWindows'),
+        (result) => assert.equal(result.value.length, 0)
+      ];
+
+      return chainPromises(steps);
+    });
+
+    it('Should return the maximum width', function() {
+      const windowTitle = 'windownamemaxwidth';
+      const maxWidth = 500;
+      const maxHeight = 600;
+      const windowOptions = getWindowOptions({
+        name: windowTitle,
+        maxWidth,
+        maxHeight
+      });
+
+      const steps = [
+        ...setupWindowSteps(windowOptions),
+        () => callAsyncWindowMethod('getMaximumSize'),
+        (result) => assert.equal(result.value[0], maxWidth)
+      ];
+
+      return chainPromises(steps);
+    });
+
+    it.skip('Should return the maximum height', function() {
+      const windowTitle = 'windownamemaxheight';
+      const maxWidth = 500;
+      const maxHeight = 600;
+      const windowOptions = getWindowOptions({
+        name: windowTitle,
+        maxWidth,
+        maxHeight
+      });
+
+      const frameSize = process.platform === 'win32' ? 20 : 25;
+
+      const steps = [
+        ...setupWindowSteps(windowOptions),
+        () => callAsyncWindowMethod('getMaximumSize'),
+        (result) => assert.equal(result.value[1], maxHeight + frameSize)
+      ];
+
+      return chainPromises(steps);
+    });
+
+    it('Should return the minimum width', function() {
+      const windowTitle = 'windownameminwidth';
+      const minWidth = 500;
+      const minHeight = 600;
+      const windowOptions = getWindowOptions({
+        name: windowTitle,
+        minWidth,
+        minHeight
+      });
+
+      const steps = [
+        ...setupWindowSteps(windowOptions),
+        () => callAsyncWindowMethod('getMinimumSize'),
+        (result) => assert.equal(result.value[0], minWidth)
+      ];
+
+      return chainPromises(steps);
+    });
+
+    it.skip('Should return the minimum height', function() {
+      const windowTitle = 'windownameminheight';
+      const minWidth = 500;
+      const minHeight = 600;
+      const windowOptions = getWindowOptions({
+        name: windowTitle,
+        minWidth,
+        minHeight
+      });
+
+      const frameSize = process.platform === 'win32' ? 20 : 25;
+
+      const steps = [
+        ...setupWindowSteps(windowOptions),
+        () => callAsyncWindowMethod('getMinimumSize'),
+        (result) => assert.equal(result.value[1], minHeight + frameSize)
+      ];
+
+      return chainPromises(steps);
+    });
+
+    it.skip('Should return the parent window', function() {
```

Comment:
From my testing it all works, however the object it returns is too large which causes the test to fail.